src/components/WalletConnect/SignMessageRequest.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  StyleSheet,
+  View,
+  ScrollView,
+  TouchableWithoutFeedback,
+  Text,
+} from 'react-native';
+import { useDispatch } from 'react-redux';
+import { useNavigation } from '@react-navigation/native';
+import { t } from 'ttag';
+import {
+  walletConnectAccept,
+  walletConnectReject
+} from '../../actions';
+import { COLORS } from '../../styles/themes';
+import NewHathorButton from '../NewHathorButton';
+import { DappContainer } from './NanoContract/DappContainer';
+import { commonStyles } from './theme';
+import { NanoContractIcon } from '../Icons/NanoContract.icon';
+
+export const SignMessageRequestData = ({ data }) => (
+  <View style={[commonStyles.card, commonStyles.cardSplit]}>
+    <View style={commonStyles.cardSplitIcon}>
+      <NanoContractIcon type='fill' color={COLORS.white} />
+    </View>
+    <View style={commonStyles.cardSplitContent}>
+      <View>
+        <Text style={styles.property}>{t`Message to sign`}</Text>
+        <Text style={styles.value}>{data.message}</Text>
+      </View>
+      <View style={commonStyles.cardSeparator} />
+      <View>
+        <Text style={styles.property}>{t`Address`}</Text>
+        <Text style={styles.value}>{data.address.address}</Text>
+      </View>
+      <View style={commonStyles.cardSeparator} />
+      <View>
+        <Text style={styles.property}>{t`Address Path`}</Text>
+        <Text style={[styles.value, styles.bold]}>{data.address.addressPath}</Text>
+      </View>
+    </View>
+  </View>
+);
+
+export const SignMessageRequest = ({ signMessageRequest }) => {
+  const { dapp, data } = signMessageRequest;
+  const { message, address } = data;
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  const onAcceptSignMessageRequest = () => {
+    const acceptedNc = { address, message };
```

Comment:
What's `Nc` in `acceptedNc`?

Reply:
"**N**o **c**opypaste" which means that I shouldn't have copy pasted it from other component

refactored it to `acceptedReq`